mlir/test/mlir-miopen-driver/populate_host.mlir

```diff
@@ -1,43 +1,150 @@
-// CHECk-NEXT: memref_cast {{.*}} : memref<?x?x?x?xf32> to memref<128x128x30x30xf32>
-// CHECk-NEXT: call @conv2d({{.*}}, {{.*}}, {{.*}}) : (memref<128x8x3x3xf32>, memref<128x8x32x32xf32>, memref<128x128x30x30xf32>) -> ()
-// CHECk-NEXT: call @mgpuMemCopy4DFloat({{.*}}, {{.*}}, {{.*}}) : (memref<?x?x?x?xf32>, memref<?x?x?x?xf32>, i32) -> ()
-// CHECk-NEXT: memref_cast {{.*}} : memref<?x?x?x?xf32> to memref<*xf32>
-// CHECk-NEXT: call @print_memref_f32({{.*}}) : (memref<*xf32>) -> ()
-// CHECk-NEXT: call @mgpuMemDealloc4DFloat({{.*}}) : (memref<?x?x?x?xf32>) -> ()
-// CHECk-NEXT: call @mgpuMemDealloc4DFloat({{.*}}) : (memref<?x?x?x?xf32>) -> ()
-// CHECk-NEXT: call @mgpuMemDealloc4DFloat({{.*}}) : (memref<?x?x?x?xf32>) -> ()
-// CHECk-NEXT: dealloc %0 : memref<128x8x3x3xf32>
-// CHECk-NEXT: dealloc %1 : memref<128x8x32x32xf32>
-// CHECk-NEXT: dealloc %2 : memref<128x128x30x30xf32>
-// CHECk-NEXT: return
-
-// CHECK-LABEL: func @conv2d(%arg0: memref<128x8x3x3xf32>, %arg1: memref<128x8x32x32xf32>, %arg2: memref<128x128x30x30xf32>)
-// CHECK-NEXT: return
+// RUN: mlir-miopen-driver -p -ph | FileCheck %s --check-prefix=F32
+// RUN: mlir-miopen-driver -p -ph -t f16 | FileCheck %s --check-prefix=F16
+// RUN: mlir-miopen-driver -p -ph -t bf16 | FileCheck %s --check-prefix=BF16
+
+// F32-LABEL: module
+// F32-NEXT: func @miopen_conv2d_kcyx_nchw_nkhw({{.*}}: memref<128x8x3x3xf32>, {{.*}}: memref<128x8x32x32xf32>, {{.*}}: memref<128x128x30x30xf32>)
+// F32-NEXT: miopen.conv2d({{.*}}, {{.*}}, {{.*}})  {arch = "{{gfx[0-9]+}}", dilations = [1 : i32, 1 : i32], filter_layout = ["k", "c", "y", "x"], input_layout = ["ni", "ci", "hi", "wi"], num_cu = {{[0-9]+}} : i32, output_layout = ["no", "ko", "ho", "wo"], padding = [0 : i32, 0 : i32], strides = [1 : i32, 1 : i32]} : memref<128x8x3x3xf32>, memref<128x8x32x32xf32>, memref<128x128x30x30xf32>
+
+// F16-LABEL: module
+// F16-NEXT: func @miopen_conv2d_kcyx_nchw_nkhw({{.*}}: memref<128x8x3x3xf16>, {{.*}}: memref<128x8x32x32xf16>, {{.*}}: memref<128x128x30x30xf16>)
+// F16-NEXT: miopen.conv2d({{.*}}, {{.*}}, {{.*}})  {arch = "{{gfx[0-9]+}}", dilations = [1 : i32, 1 : i32], filter_layout = ["k", "c", "y", "x"], input_layout = ["ni", "ci", "hi", "wi"], num_cu = {{[0-9]+}} : i32, output_layout = ["no", "ko", "ho", "wo"], padding = [0 : i32, 0 : i32], strides = [1 : i32, 1 : i32]} : memref<128x8x3x3xf16>, memref<128x8x32x32xf16>, memref<128x128x30x30xf16>
+
+// BF16-LABEL: module
+// BF16-NEXT: func @miopen_conv2d_kcyx_nchw_nkhw({{.*}}: memref<128x8x3x3xbf16>, {{.*}}: memref<128x8x32x32xbf16>, {{.*}}: memref<128x128x30x30xbf16>)
+// BF16-NEXT: miopen.conv2d({{.*}}, {{.*}}, {{.*}})  {arch = "{{gfx[0-9]+}}", dilations = [1 : i32, 1 : i32], filter_layout = ["k", "c", "y", "x"], input_layout = ["ni", "ci", "hi", "wi"], num_cu = {{[0-9]+}} : i32, output_layout = ["no", "ko", "ho", "wo"], padding = [0 : i32, 0 : i32], strides = [1 : i32, 1 : i32]} : memref<128x8x3x3xbf16>, memref<128x8x32x32xbf16>, memref<128x128x30x30xbf16>
```

Comment:
I think we should get rid of this. This section has already been tested in `mlir/test/mlir-miopen-driver/populate.mlir`. By having this section we are testing whether two options can be used at the same time. That should already be covered at command line argument parser.

Reply:
will cover this in a subsequent PR.